src/packages/core/property-editor/uis/tree-picker/config/filter/property-editor-ui-tree-picker-filter.element.ts

```diff
@@ -0,0 +1,95 @@
+import { UMB_DATA_TYPE_WORKSPACE_CONTEXT } from '../../../../../data-type/workspace/data-type-workspace.context.js';
+import { UmbDocumentTypeInputElement } from '../../../../../../documents/document-types/components/document-type-input/document-type-input.element.js';
+import { UmbMediaTypeInputElement } from '../../../../../../media/media-types/components/media-type-input/media-type-input.element.js';
+import { StartNode } from '@umbraco-cms/backoffice/components';
+import type { UmbPropertyEditorUiElement } from '@umbraco-cms/backoffice/extension-registry';
+import { customElement, html, property } from '@umbraco-cms/backoffice/external/lit';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
+
+@customElement('umb-property-editor-ui-tree-picker-filter')
+export class UmbPropertyEditorUITreePickerFilterElement extends UmbLitElement implements UmbPropertyEditorUiElement {
+	constructor() {
+		super();
+
+		this.consumeContext(UMB_DATA_TYPE_WORKSPACE_CONTEXT, (instance) => {
+			const workspace = instance;
+			this.observe(workspace.data, (data) => {
+				const property = data?.values.find((setting) => setting.alias === 'startNode');
+				if (property) {
+					const startNode = property.value as StartNode;
+					if (startNode.type) {
+						this.sourceType = startNode.type;
```

Comment:
If the source type changes, then we need to make sure we reset the value. So we dont carry over ex. Document-Types to the Media-Types picker.

One solution I see to this, would be to not preset the sourceType to content, but have it as undefined initially.

then you could make this code here:

```
    const oldSourceType = this.sourceType;
    this.sourceType = startNode.type;
    // If we has a sourceType before, we can see this as a change and not the initial value, so lets reset the value, so we dont carry over content-types to the new source type.
    if(oldSourceType && oldSourceType !== this.sourceType) {
        this.value = [];
    }
```

Reply:
I've added this in now. 👍